label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRule data = 2;
+}
+
+message CreateLabelApplicationRule {
+  LabelApplicationRule data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  string description = 2;
+  MatchingCriteria matching_criteria = 3;
+  ApplicationCriteria application_criteria = 4;
+
+  message MatchingCriteria {
+    Type type = 1;
+    repeated Condition conditions = 2;
+
+    enum Type {
+      TYPE_UNSPECIFIED = 0;
+      TYPE_SPAN = 1;
+      TYPE_TRACE = 2;
+    }
+
+    message Condition {
+      string name = 1;
+      Scope scope = 2;
+      string key = 3;
+      Operator operator = 4;
+      optional string value = 5;
```

Comment:
Can values be of different data types?

Reply:
I thought it is simplest to start with string data type. We can add support for more data types for values as and when needed. It will make out operator support richer. For numbers, we can support `<`, `>`  etc. 

What do you say? Or shall we add support for numbers right now?

Reply:
We don't need to add number support right now, but it gets at the design evolution. The operators for strings and numbers are different (as are for unary, no-arg operators like `exists`, for that matter). So if we were to add number support, would we be doing something like a oneof with different messages encompassing the string operators/value, numeric operators/value, unary etc?) I'm less concerned with functionality than compatibility.

```proto
message Condition {
  oneof condition {
    StringCondition string_condition = 1;
    NumericCondition numeric_condition = 2;
    UnaryCondition unary_condition = 3;
  }
}
  ```

Reply:
Makes sense. Updated accordingly